Fix: Prefs test suite

ranjit-exalt opened the pull request:
Applying this patch will fix the following seven test cases of "prefs" suite: 

1. ff_can_be_set_always_private.py (#4531)
2. ff_can_be_set_clear_data_on_exit.py (#4535)
3. ff_can_be_set_never_remember_history.py(#4536)
4. remove_selected_button_grayed_out_after_site_deselected.py(#)
5. several_custom_settings_remains_checked.py(#4505)
6. switch_immediately_opened_link.py(#4573)
7. ff_select_which_highlights_to_display.py(#4486)

Commit:
Applying this patch will fix following seven test cases of "prefs" test suite:

1. ff_can_be_set_always_private.py
2. ff_can_be_set_clear_data_on_exit.py
3. ff_can_be_set_never_remember_history.py
4. remove_selected_button_grayed_out_after_site_deselected.py
5. several_custom_settings_remains_checked.py
6. switch_immediately_opened_link.py
7. ff_select_which_highlights_to_display.py

spronoza reviewed (changes requested):
Failing on OSX
![image](https://user-images.githubusercontent.com/46671906/72908066-a217f580-3d3d-11ea-852c-9bc663463cef.png)

Commit:
Change log for "ff_can_be_set_clear_data_on_exit.py":
1. Added firefox wait time for exists().

Change log for "ff_select_which_highlights_to_display.py":
1. Added 2 second interval in key press so that in slow systems, pop-up get sometime to appear.

Commit:
Merged dev to issue_4531 to resolve conflict.
